### DISCUSSION TITLE
how to update many?

### DISCUSSION THREAD

**Discussion by wesleytoshio** [simolus3/drift#1157] [General]
I have a list of selected items eg: [item1, item2, item3] and I need to update them all at once, any example that can help me?
```dart
//This exemplo update one
Future updateNote(Insertable<Note> note) => update(notes).replace(note);
```

**Comment by simolus3** [comment]
Instead of using `replace` (which targets a single row), you can just use `write` and `where` to update some rows. The documentation has an example [here](https://moor.simonbinder.eu/docs/getting-started/writing_queries/#updates-and-deletes), does that help?

In your case, it would probably look like this:

```dart
(update(notes)..where((note) => note.id.isIn([your, target, ids]))).write(yourUpdate)
```

**Reply by wesleytoshio** [reply]
in this case, what would be [your] and [target]?

**Reply by wesleytoshio** [reply]
I need to change the value of several items that were selected in my list eg: checked
````dart
Future updateSelected(List<Note> selecteds){
// ex:
(update(notes)..where((note) => note.id.isIn(selecteds.map((e) => e.id).toList())))
          .write(selecteds);
}
````

**Reply by simolus3** [reply]
I just meant that `[your, target, ids]` would be a list of ids to update, but that would apply the same update to every row. Judging by your second snippet, it looks like you essentially want to perform many independent updates? In that case, you can use a batch:

```dart
await batch((b) {
  b.replaceAll(notes, selecteds);
});
```
